ocaml/typing/subst.ml

```diff
@@ -474,6 +474,7 @@ let type_declaration' copy_scope s decl =
             prepare_jkind decl.type_loc decl.type_jkind
         | Duplicate_variables | No_action -> decl.type_jkind
       end;
+    type_jkind_annotation = decl.type_jkind_annotation;
```

Comment:
`(* CR layouts v10: Apply the substitution here, too *)`

(v10 might not be all that far off -- once we have v2.8 done, we can do abstract kinds, and we might want to to simplify interfaces)